src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaInstructorTypeService.java

```diff
@@ -1,73 +1,57 @@
+import edu.ucdavis.dss.ipa.entities.UserRole;
 import edu.ucdavis.dss.ipa.repositories.InstructorTypeRepository;
-import edu.ucdavis.dss.ipa.services.InstructorCostService;
 import edu.ucdavis.dss.ipa.services.InstructorTypeService;
+import edu.ucdavis.dss.ipa.services.UserRoleService;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 
 import javax.inject.Inject;
 import java.util.List;
 
 @Service
 public class JpaInstructorTypeService implements InstructorTypeService {
-    @Inject InstructorTypeRepository instructorTypeRepository;
-    @Inject InstructorCostService instructorCostService;
+    static long INSTRUCTOR_TYPE = 7L;
 
-    @Override
-    public List<InstructorType> findByBudgetId(Long budgetId) {
-        return instructorTypeRepository.findByBudgetId(budgetId);
-    }
+    @Inject InstructorTypeRepository instructorTypeRepository;
+    @Inject UserRoleService userRoleService;
 
     @Override
     public InstructorType findById(Long instructorTypeId) {
         return instructorTypeRepository.findById(instructorTypeId);
     }
 
     @Override
-    @Transactional
-    public void deleteById(long instructorTypeId) {
-        InstructorType instructorType = this.findById(instructorTypeId);
-
-        instructorCostService.removeAssociationByInstructorTypeId(instructorTypeId);
-
-        instructorTypeRepository.deleteById(instructorTypeId);
+    public List<InstructorType> getAllInstructorTypes() {
+        return (List<InstructorType>) instructorTypeRepository.findAll();
     }
 
+    /**
+     * Attempt to find the relevant instructorType for the instructor based on userRoles. Will return 'instructor' as a default type if none are found.
+     * @param instructor
+     * @param schedule
+     * @return
+     */
     @Override
-    public InstructorType update(InstructorType newInstructorType) {
-        InstructorType originalInstructorType = this.findById(newInstructorType.getId());
-
-        if(originalInstructorType == null) {
-            return null;
-        }
-
-        originalInstructorType.setCost(newInstructorType.getCost());
-        originalInstructorType.setDescription(newInstructorType.getDescription());
+    public InstructorType findByInstructorAndSchedule(Instructor instructor, Schedule schedule) {
+        List<UserRole> userRoles = userRoleService.findByLoginIdAndWorkgroup(instructor.getLoginId(), schedule.getWorkgroup());
 
-        return this.instructorTypeRepository.save(originalInstructorType);
-    }
+        InstructorType instructorType = null;
 
-    @Override
-    public InstructorType findOrCreate(InstructorType instructorTypeDTO) {
-        if (instructorTypeDTO == null || instructorTypeDTO.getBudget() == null) {
-            return null;
+        for (UserRole userRole : userRoles) {
+            if (userRole.getRoleToken().equals("instructor")) {
+                instructorType = userRole.getInstructorType();
```

Comment:
This only takes the last instructor type. If you only care about one, you should probably break after setting instructorType.